kernel/src/table_changes/scan_file.rs

```diff
@@ -0,0 +1,447 @@
+                continue;
+            }
+            let timestamp = getters[16].get(row_index, "scanFile.timestamp")?;
+            let commit_version: i64 = getters[17].get(row_index, "scanFile.commit_version")?;
+            let commit_version: u64 = commit_version.try_into().map_err(Error::generic)?;
+
+            if let Some(path) = getters[0].get_opt(row_index, "scanFile.add.path")? {
+                let deletion_vector = visit_deletion_vector_at(row_index, &getters[1..=5])?;
+                let dv_info = DvInfo { deletion_vector };
+                let partition_values =
+                    getters[6].get(row_index, "scanFile.add.fileConstantValues.partitionValues")?;
+                let scan_file = CDFScanFile {
+                    scan_type: CDFScanFileType::Add,
+                    path,
+                    dv_info,
+                    partition_values,
+                    commit_version,
+                    timestamp,
+                };
+                (self.callback)(&mut self.context, scan_file)
+            } else if let Some(path) = getters[7].get_opt(row_index, "scanFile.remove.path")? {
+                let deletion_vector = visit_deletion_vector_at(row_index, &getters[8..=12])?;
+                let dv_info = DvInfo { deletion_vector };
+                let partition_values = getters[13].get(
+                    row_index,
+                    "scanFile.remove.fileConstantValues.partitionValues",
+                )?;
+                let scan_file = CDFScanFile {
+                    scan_type: CDFScanFileType::Remove,
+                    path,
+                    dv_info,
+                    partition_values,
+                    commit_version,
+                    timestamp,
+                };
+                (self.callback)(&mut self.context, scan_file)
+            } else if let Some(path) = getters[14].get_opt(row_index, "scanFile.cdc.path")? {
+                let partition_values = getters[15]
+                    .get(row_index, "scanFile.cdc.fileConstantValues.partitionValues")?;
+                let scan_file = CDFScanFile {
+                    scan_type: CDFScanFileType::Cdc,
+                    path,
+                    dv_info: DvInfo {
+                        deletion_vector: None,
+                    },
+                    partition_values,
+                    commit_version,
+                    timestamp,
+                };
+
+                (self.callback)(&mut self.context, scan_file)
+            }
+        }
+        Ok(())
+    }
+
+    fn selected_column_names_and_types(&self) -> (&'static [ColumnName], &'static [DataType]) {
+        static NAMES_AND_TYPES: LazyLock<ColumnNamesAndTypes> =
+            LazyLock::new(|| cdf_scan_row_schema().leaves(None));
+        NAMES_AND_TYPES.as_ref()
+    }
+}
+
+/// Get the schema that scan rows (from [`TableChanges::scan_data`]) will be returned with.
+///
+/// It is:
+/// ```ignored
+/// {
+///   add: {
+///     path: string,
```

Comment:
I'm torn whether we should include this comment... it's definitely easier to read than the raw schema manipulation, but it's also super prone to not get updated as the schema evolves over time. And an inaccurate comment is worse than no comment at all.

Reply:
Hmm that's what exists in the scan code, but I agree that this can be tricky to keep up to date.

Given the way schema is initialized, it's not hard to read the code. iirc at some point we used to initialize all the subfields in go without splitting them up into variables. That may be why we introduced this comment.

So +1 to removing the comment. I'll make that change over in the other PR.